dart/lib/src/annotations.dart

```diff
@@ -69,6 +69,25 @@ class ByClass implements Finder {
   String toString() => '@ByClass("$className")';
 }
 
+/// Finds elements with the given tag name. Unlike [ByTagName],
+/// this will also find the current Root if it is the given tag.
+class EnsureTag implements Finder {
+  final String tagName;
+
+  const EnsureTag(this.tagName);
+
+  @override
+  List<PageLoaderElement> findElements(PageLoaderElement context) {
+    Union union = new Union([Root, new ByTagName(this.tagName)]);
```

Comment:
This can be written more efficiently:

```
  List<PageLoaderElement> elements = [];
  if (context.name == tagName) {
    elements.add(context);
  }
  elements.addAll(context.getElementsByCss(this.tagName));
  return new UnmodifiableListView<PageLoaderElement>(elements);
}